SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -257,44 +258,72 @@ Worker_ComponentUpdate InterestFactory::CreateInterestUpdate() const
 	return CreateInterest().CreateInterestUpdate();
 }
 
-Interest InterestFactory::CreateServerWorkerInterest()
+Interest InterestFactory::CreateServerWorkerInterest(const UAbstractLBStrategy* LBStrategy)
 {
-	QueryConstraint Constraint;
-
 	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
 	if (SpatialGDKSettings->bEnableServerQBI)
 	{
 		UE_LOG(LogInterestFactory, Warning, TEXT("For performance reasons, it's recommended to disable server QBI"));
 	}
 
-	if (!SpatialGDKSettings->bEnableServerQBI && SpatialGDKSettings->bEnableOffloading)
+	Interest ServerInterest;
+	ComponentInterest ServerComponentInterest;
+	Query ServerQuery;
+	QueryConstraint Constraint;
+
+	// Set the result type of the query
+	if (SpatialGDKSettings->bEnableResultTypes)
 	{
-		// In offloading scenarios, hijack the server worker entity to ensure each server has interest in all entities
-		Constraint.ComponentConstraint = SpatialConstants::POSITION_COMPONENT_ID;
+		ServerQuery.ResultComponentId = ServerNonAuthInterestResultType;
 	}
 	else
 	{
-		// Ensure server worker receives always relevant entities
-		Constraint = CreateAlwaysRelevantConstraint();
+		ServerQuery.FullSnapshotResult = true;
 	}
 
-	Query Query;
-	Query.Constraint = Constraint;
-	if (SpatialGDKSettings->bEnableResultTypes)
-	{
-		Query.ResultComponentId = ServerNonAuthInterestResultType;
-	}
-	else
+	if (!SpatialGDKSettings->bEnableServerQBI && SpatialGDKSettings->bEnableOffloading)
 	{
-		Query.FullSnapshotResult = true;
+		// In offloading scenarios, hijack the server worker entity to ensure each server has interest in all entities
+		Constraint.ComponentConstraint = SpatialConstants::POSITION_COMPONENT_ID;
+		ServerQuery.Constraint = Constraint;
+
+		// No need to add any further interest as we are already interested in everything
+		AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::POSITION_COMPONENT_ID, ServerQuery);
+		return ServerInterest;
 	}
 
-	ComponentInterest Queries;
-	Queries.Queries.Add(Query);
+	// If we aren't offloading, the server gets more granular interest.
 
-	Interest ServerInterest;
-	ServerInterest.ComponentInterestMap.Add(SpatialConstants::POSITION_COMPONENT_ID, Queries);
+	// Ensure server worker receives always relevant entities
+	QueryConstraint AlwaysRelevantConstraint = CreateAlwaysRelevantConstraint();
+
+	Constraint = AlwaysRelevantConstraint;
+
+	// If we aren't using the load balancer, there is no need for further interest as we are either offloading or single server,
+	// so will checkout everything we care about through authority.
+	if (SpatialGDKSettings->bEnableUnrealLoadBalancer && SpatialGDKSettings->bEnableServerQBI)
```

Comment:
Maybe we don't need `bEnableServerQBI`, and we can just use `bEnableUnrealLoadBalancer` instead?

Reply:
This is a great idea, happy to do it. In this PR or a follow up?

Reply:
My preference is this PR, but will defer to you.

Reply:
alright alright

Reply:
flag gone